Unable to build via Docker, issue with M1 Macs?

Are there known issues with M1 Macs?

I tried building via Docker, both in the current `master` branch and in `6.6.1`.

I am following the instructions exactly as written. Download the source, `cd docker` and `docker-composer up -d`

All the endpoints and `/status` gives 404s. Though I am able to access the Tomcat page at `localhost:8080`.

This is the error log:

```
### openrouteservice configuration ###

No ors-config.json in ors-conf folder. Copy config from /ors-core/openrouteservice/src/main/resources/ors-config.json

### Package openrouteservice and deploy to Tomcat ###

WARNING: An illegal reflective access operation has occurred

WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)

WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1

WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations

WARNING: All illegal access operations will be denied in a future release

9215 [WARNING] Error injecting: org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18

java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

    at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:143)

    at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)

    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)

    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)

    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)

    at org.eclipse.sisu.bean.BeanScheduler$CycleActivator.onProvision (BeanScheduler.java:230)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:120)

    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)

    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)

    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get (ProviderToInternalFactoryAdapter.java:40)

    at com.google.inject.internal.SingletonScope$1.get (SingletonScope.java:168)

    at com.google.inject.internal.InternalFactoryToProviderAdapter.get (InternalFactoryToProviderAdapter.java:39)

    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)

    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)

    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)

    at java.util.AbstractMap.get (AbstractMap.java:187)

    at org.apache.maven.lifecycle.internal.DefaultLifecyclePluginAnalyzer.getPluginsBoundByDefaultToAllLifecycles (DefaultLifecyclePluginAnalyzer.java:90)

    at org.apache.maven.model.plugin.DefaultLifecycleBindingsInjector.injectLifecycleBindings (DefaultLifecycleBindingsInjector.java:64)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:462)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:583)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:373)

    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

Caused by: java.lang.ClassNotFoundException: org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18

    at org.eclipse.sisu.space.CloningClassSpace$CloningClassLoader.loadClass (CloningClassSpace.java:122)

    at java.lang.ClassLoader.loadClass (ClassLoader.java:522)

    at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:139)

    at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)

    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)

    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)

    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)

    at org.eclipse.sisu.bean.BeanScheduler$CycleActivator.onProvision (BeanScheduler.java:230)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:120)

    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)

    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)

    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get (ProviderToInternalFactoryAdapter.java:40)

    at com.google.inject.internal.SingletonScope$1.get (SingletonScope.java:168)

    at com.google.inject.internal.InternalFactoryToProviderAdapter.get (InternalFactoryToProviderAdapter.java:39)

    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)

    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)

    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)

    at java.util.AbstractMap.get (AbstractMap.java:187)

    at org.apache.maven.lifecycle.internal.DefaultLifecyclePluginAnalyzer.getPluginsBoundByDefaultToAllLifecycles (DefaultLifecyclePluginAnalyzer.java:90)

    at org.apache.maven.model.plugin.DefaultLifecycleBindingsInjector.injectLifecycleBindings (DefaultLifecycleBindingsInjector.java:64)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:462)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:583)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:373)

    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

9386 [ERROR] Internal error: com.google.inject.ProvisionException: Unable to provision, see the following errors:

9389 [ERROR] 

9390 [ERROR] 1) Error in custom provider, java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

9391 [ERROR]   at ClassRealm[plexus.core, parent: null] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)

9392 [ERROR]   while locating org.apache.maven.lifecycle.mapping.LifecycleMapping annotated with @com.google.inject.name.Named(value="war")

9393 [ERROR] 

9394 [ERROR] 1 error: Unable to provision, see the following errors:

9394 [ERROR] 

9395 [ERROR] 1) Error in custom provider, java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

9396 [ERROR]   at ClassRealm[plexus.core, parent: null] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)

9398 [ERROR]   while locating org.apache.maven.lifecycle.mapping.LifecycleMapping annotated with @com.google.inject.name.Named(value="war")

9399 [ERROR] 

9399 [ERROR] 1 error: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present: org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18

9402 [ERROR] -> [Help 1]

org.apache.maven.InternalErrorException: Internal error: com.google.inject.ProvisionException: Unable to provision, see the following errors:


1) Error in custom provider, java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

  at ClassRealm[plexus.core, parent: null] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)

  while locating org.apache.maven.lifecycle.mapping.LifecycleMapping annotated with @com.google.inject.name.Named(value="war")


1 error

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

Caused by: com.google.inject.ProvisionException: Unable to provision, see the following errors:


1) Error in custom provider, java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

  at ClassRealm[plexus.core, parent: null] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)

  while locating org.apache.maven.lifecycle.mapping.LifecycleMapping annotated with @com.google.inject.name.Named(value="war")


1 error

    at com.google.inject.internal.InternalProvisionException.toProvisionException (InternalProvisionException.java:226)

    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1053)

    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)

    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)

    at java.util.AbstractMap.get (AbstractMap.java:187)

    at org.apache.maven.lifecycle.internal.DefaultLifecyclePluginAnalyzer.getPluginsBoundByDefaultToAllLifecycles (DefaultLifecyclePluginAnalyzer.java:90)

    at org.apache.maven.model.plugin.DefaultLifecycleBindingsInjector.injectLifecycleBindings (DefaultLifecycleBindingsInjector.java:64)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:462)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:583)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:373)

    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

Caused by: java.lang.TypeNotPresentException: Type org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18 not present

    at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:143)

    at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)

    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)

    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)

    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)

    at org.eclipse.sisu.bean.BeanScheduler$CycleActivator.onProvision (BeanScheduler.java:230)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:120)

    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)

    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)

    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get (ProviderToInternalFactoryAdapter.java:40)

    at com.google.inject.internal.SingletonScope$1.get (SingletonScope.java:168)

    at com.google.inject.internal.InternalFactoryToProviderAdapter.get (InternalFactoryToProviderAdapter.java:39)

    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)

    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)

    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)

    at java.util.AbstractMap.get (AbstractMap.java:187)

    at org.apache.maven.lifecycle.internal.DefaultLifecyclePluginAnalyzer.getPluginsBoundByDefaultToAllLifecycles (DefaultLifecyclePluginAnalyzer.java:90)

    at org.apache.maven.model.plugin.DefaultLifecycleBindingsInjector.injectLifecycleBindings (DefaultLifecycleBindingsInjector.java:64)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:462)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:583)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:373)

    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

Caused by: java.lang.ClassNotFoundException: org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping$__sisu18

    at org.eclipse.sisu.space.CloningClassSpace$CloningClassLoader.loadClass (CloningClassSpace.java:122)

    at java.lang.ClassLoader.loadClass (ClassLoader.java:522)

    at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:139)

    at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)

    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)

    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)

    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)

    at org.eclipse.sisu.bean.BeanScheduler$CycleActivator.onProvision (BeanScheduler.java:230)

    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:120)

    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)

    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)

    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)

    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get (ProviderToInternalFactoryAdapter.java:40)

    at com.google.inject.internal.SingletonScope$1.get (SingletonScope.java:168)

    at com.google.inject.internal.InternalFactoryToProviderAdapter.get (InternalFactoryToProviderAdapter.java:39)

    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)

    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)

    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)

    at java.util.AbstractMap.get (AbstractMap.java:187)

    at org.apache.maven.lifecycle.internal.DefaultLifecyclePluginAnalyzer.getPluginsBoundByDefaultToAllLifecycles (DefaultLifecyclePluginAnalyzer.java:90)

    at org.apache.maven.model.plugin.DefaultLifecycleBindingsInjector.injectLifecycleBindings (DefaultLifecycleBindingsInjector.java:64)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:462)

    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:583)

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:373)

    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

9536 [ERROR] 

9537 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.

9538 [ERROR] Re-run Maven using the -X switch to enable full debug logging.

9538 [ERROR] 

9538 [ERROR] For more information about the errors and possible solutions, please read the following articles:

9540 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException

NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED

05-Oct-2021 18:11:21.428 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version:        Apache Tomcat/8.5.39

05-Oct-2021 18:11:21.435 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Mar 14 2019 11:24:26 UTC

05-Oct-2021 18:11:21.436 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server number:         8.5.39.0

05-Oct-2021 18:11:21.438 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux

05-Oct-2021 18:11:21.439 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            5.10.47-linuxkit

05-Oct-2021 18:11:21.441 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64

05-Oct-2021 18:11:21.442 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/local/openjdk-11

05-Oct-2021 18:11:21.443 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           11.0.12+7

05-Oct-2021 18:11:21.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Oracle Corporation

05-Oct-2021 18:11:21.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/local/tomcat

05-Oct-2021 18:11:21.446 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/local/tomcat

05-Oct-2021 18:11:21.461 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED

05-Oct-2021 18:11:21.463 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED

05-Oct-2021 18:11:21.465 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED

05-Oct-2021 18:11:21.466 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties

05-Oct-2021 18:11:21.466 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager

05-Oct-2021 18:11:21.468 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.awt.headless=true

05-Oct-2021 18:11:21.469 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:TargetSurvivorRatio=75

05-Oct-2021 18:11:21.469 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:SurvivorRatio=64

05-Oct-2021 18:11:21.470 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:MaxTenuringThreshold=3

05-Oct-2021 18:11:21.471 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:+UseG1GC

05-Oct-2021 18:11:21.472 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:+ScavengeBeforeFullGC

05-Oct-2021 18:11:21.473 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -XX:ParallelGCThreads=4

05-Oct-2021 18:11:21.473 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xms1g

05-Oct-2021 18:11:21.474 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xmx2g

05-Oct-2021 18:11:21.475 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048

05-Oct-2021 18:11:21.476 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources

05-Oct-2021 18:11:21.478 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027

05-Oct-2021 18:11:21.480 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcom.sun.management.jmxremote

05-Oct-2021 18:11:21.481 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcom.sun.management.jmxremote.port=9001

05-Oct-2021 18:11:21.482 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcom.sun.management.jmxremote.rmi.port=9001

05-Oct-2021 18:11:21.482 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcom.sun.management.jmxremote.authenticate=false

05-Oct-2021 18:11:21.483 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcom.sun.management.jmxremote.ssl=false

05-Oct-2021 18:11:21.483 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.rmi.server.hostname=localhost

05-Oct-2021 18:11:21.484 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dignore.endorsed.dirs=

05-Oct-2021 18:11:21.485 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/local/tomcat

05-Oct-2021 18:11:21.485 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/local/tomcat

05-Oct-2021 18:11:21.486 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp

05-Oct-2021 18:11:21.487 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: [/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib]

05-Oct-2021 18:11:21.646 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]

05-Oct-2021 18:11:21.695 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read

05-Oct-2021 18:11:21.779 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["ajp-nio-8009"]

05-Oct-2021 18:11:21.787 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read

05-Oct-2021 18:11:21.793 INFO [main] org.apache.catalina.startup.Catalina.load Initialization processed in 1675 ms

05-Oct-2021 18:11:22.116 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]

05-Oct-2021 18:11:22.117 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet Engine: Apache Tomcat/8.5.39

05-Oct-2021 18:11:22.174 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/examples]

05-Oct-2021 18:11:24.971 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/examples] has finished in [2,792] ms

05-Oct-2021 18:11:24.997 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/manager]

05-Oct-2021 18:11:25.197 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/manager] has finished in [200] ms

05-Oct-2021 18:11:25.198 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/docs]

05-Oct-2021 18:11:25.282 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/docs] has finished in [84] ms

05-Oct-2021 18:11:25.283 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/host-manager]

05-Oct-2021 18:11:25.381 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/host-manager] has finished in [97] ms

05-Oct-2021 18:11:25.381 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/ROOT]

05-Oct-2021 18:11:25.462 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/ROOT] has finished in [81] ms

05-Oct-2021 18:11:25.504 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]

05-Oct-2021 18:11:25.568 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-8009"]

05-Oct-2021 18:11:25.587 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 3792 ms
```

--- a/.github/workflows/build-and-publish.yml
+++ b/.github/workflows/build-and-publish.yml
@@ -9,7 +9,7 @@ on:
 env:
   TEST_IMAGE_NAME: 'local/openrouteservice:test'
   PRODUCTION_IMAGE_NAME: 'openrouteservice/openrouteservice:nightly'
-  BUILD_PLATFORMS: 'linux/amd64'
+  BUILD_PLATFORMS: 'linux/amd64,linux/arm64'
 
 
 jobs:
@@ -23,19 +23,60 @@ jobs:
       build_platforms: ${{ env.BUILD_PLATFORMS }}
     steps:
       - run: |
-          echo "Publish enironment variables"
-  build_docker_and_test_war_deployment:
-    name: Build and check the war deployment stage
+          echo "Publish environment variables"
+  build_docker_images:
+    name: Build the docker images
     runs-on: ubuntu-latest
     needs:
       - prepare_environment
     steps:
       - name: Checkout
-        uses: actions/checkout@v2.2.0
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+      - name: Build image for platforms ${{ needs.prepare_environment.outputs.build_platforms }}
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: false
+          load: false
+          tags: ${{ needs.prepare_environment.outputs.test_image_name }}
+          build-args: "--platform ${{ needs.prepare_environment.outputs.build_platforms }}"
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+  run_docker_image_tests:
+    name: Run & test ${{ matrix.arch }}
+    runs-on: ubuntu-latest
+    needs:
+      - prepare_environment
+      - build_docker_images
+    strategy:
+      matrix:
+        include:
+          - arch: amd64
+            platform: linux/amd64
+            image: ubuntu-latest
+    #          - arch: arm64
+    #            platform: linux/arm64
+    #            image: ubuntu-latest
+    steps:
+      - run: |
+          echo "Run docker test for platform ${{ matrix.arch }}"
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up QEMU for ${{ matrix.arch }}
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ${{ matrix.platform }}
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
         id: buildx
@@ -53,10 +94,10 @@ jobs:
           push: false
           load: true
           tags: ${{ needs.prepare_environment.outputs.test_image_name }}
-          build-args: "--platform ${{ needs.prepare_environment.outputs.build_platforms }}"
+          build-args: "--platform ${{ matrix.platform }}"
           cache-from: type=gha
           cache-to: type=gha,mode=max
-      - name: Start container from image and wait for successful checks
+      - name: Start container from previously build image and wait for successful checks
         run: |
           mkdir $(pwd)/graphs $(pwd)/conf
           sudo chown 1000:1000 $(pwd)/graphs $(pwd)/conf
@@ -78,8 +119,7 @@ jobs:
     name: Publish the docker image to docker hub
     runs-on: ubuntu-latest
     needs:
-      - prepare_environment
-      - build_docker_and_test_war_deployment
+      - run_docker_image_tests
     steps:
       - name: Checkout
         uses: actions/checkout@v2.2.0

--- a/.github/workflows/build-and-publish.yml
+++ b/.github/workflows/build-and-publish.yml
@@ -63,9 +63,9 @@ jobs:
           - arch: amd64
             platform: linux/amd64
             image: ubuntu-latest
-    #          - arch: arm64
-    #            platform: linux/arm64
-    #            image: ubuntu-latest
+          - arch: arm64
+            platform: linux/arm64
+            image: ubuntu-latest
     steps:
       - run: |
           echo "Run docker test for platform ${{ matrix.arch }}"

